rcgen/src/certificate.rs

```diff
@@ -0,0 +1,1496 @@
+use std::net::IpAddr;
+use std::str::FromStr;
+
+#[cfg(feature = "pem")]
+use pem::Pem;
+use time::{Date, Month, OffsetDateTime, PrimitiveDateTime, Time};
+use yasna::models::ObjectIdentifier;
+use yasna::{DERWriter, Tag};
+
+use crate::crl::CrlDistributionPoint;
+use crate::csr::CertificateSigningRequestParams;
+use crate::key_pair::PublicKeyData;
+#[cfg(feature = "crypto")]
+use crate::ring_like::digest;
+#[cfg(feature = "pem")]
+use crate::ENCODE_CONFIG;
+use crate::{
+	write_distinguished_name, write_dt_utc_or_generalized, write_x509_authority_key_identifier,
+	write_x509_extension, DistinguishedName, Error, KeyIdMethod, KeyPair, KeyUsagePurpose, SanType,
+	SerialNumber, OID_BASIC_CONSTRAINTS, OID_COMMON_NAME, OID_COUNTRY_NAME,
+	OID_CRL_DISTRIBUTION_POINTS, OID_EXT_KEY_USAGE, OID_KEY_USAGE, OID_LOCALITY_NAME,
```

Comment:
Imports with these many lines are an anti-pattern as their interaction with rustfmt and text reflow is prone to cause merge conflicts. If you removed the `OID_` prefix from the oids and moved all the `OID_` related stuff to a separate `mod oid`, contained inside `lib.rs`, and then used the OIDs e.g. via `oid::BASIC_CONSTRAINTS`, then the number of lines of the import would be greatly reduced.

Alternatively, one could have a star import `use crate::oid::*`. I generally dislike star imports but if everything in the module is prefixed with the `OID_` name then it's pretty clear where the item is from.

Reply:
Good call! There was actually already an `oid` module, I just failed to notice it. Imported the module only in the new `certificate` module to reduce the number of imports here.